solver/src/driver.rs

```diff
@@ -312,19 +287,21 @@ impl Driver {
                 .expect("Price of native token must be known.");
 
         futures::stream::iter(settlements)
-            .filter_map(|settlement| async {
-                let surplus = settlement.settlement.total_surplus(prices);
+            .filter_map(|settlement_with_solver_id| async {
+                let (solver_id, settlement) = settlement_with_solver_id;
+                let surplus = settlement.total_surplus(prices);
                 // Because of a potential fee discount, the solver fees may by themselves not be sufficient to make a solution economically viable (leading to a negative objective value)
                 // We therefore reverse apply the fee discount to simulate unsubsidized fees for ranking.
-                let unsubsidized_solver_fees = settlement.settlement.total_fees(prices) / BigRational::from_float(self.fee_discount_factor).expect("Discount factor is not a rational");
+                let unsubsidized_solver_fees = settlement.total_fees(prices) / BigRational::from_float(self.fee_discount_factor).expect("Discount factor is not a rational");
+                // TODO: can we reuse gas estimate from simulation?
```

Comment:
At this moment I can't see why not. It would reduce our eth calls and possibly eliminate some of these

```
ERROR solver::driver: single run errored: failed to estimate gas price
Caused by:
   all gas estimators failed
```

Reply:
That error is from the gas price estimator but what is estimated here is the gas use. 
The gas use could probably be taken from the existing simulation. The problem is that the simulation at the moment uses "view methods" which don't return the gas used while the gas estimation uses the "estimate gas" rpc call which does. If the latter also checks for reverts just like view methods but in addition returns used gas then we could use it instead.